File: app/src/main/java/net/feheren_fekete/applist/applistpage/viewmodel/BaseItem.java
package net.feheren_fekete.applist.applistpage.viewmodel;

public abstract class BaseItem {

    private long mId;
    private String mName;
    private boolean mIsEnabled = true;
    private boolean mIsHighlighted = false;
    private boolean mIsSelected = false;
    private boolean mIsDragged = false;

    public BaseItem(long id, String name) {
        mId = id;
        mName = name;
    }

    public long getId() {
        return mId;
    }

    public boolean isEnabled() {
        return mIsEnabled;
    }

    public void setEnabled(boolean enabled) {
        mIsEnabled = enabled;
    }

    public boolean isDragged() {
        return mIsDragged;
    }

    public void setDragged(boolean dragged) {
        mIsDragged = dragged;
    }

    public boolean isSelected() {
        return mIsSelected;
    }

    public void setSelected(boolean selected) {
        mIsSelected = selected;
    }

    public boolean isHighlighted() {
        return mIsHighlighted;
    }

    public void setHighlighted(boolean highlighted) {
        mIsHighlighted = highlighted;
    }

    public String getName() {
        return mName;
    }
}

File: app/src/main/java/net/feheren_fekete/applist/applistpage/viewmodel/BaseItem.kt
package net.feheren_fekete.applist.applistpage.viewmodel

abstract class BaseItem(val id: Long, val name: String) {

    var isEnabled = true

    var isDraggedOverLeft = false
        private set

    var isDraggedOverRight = false
        private set

    var isHighlighted = false

    fun setDraggedOverState(state: DraggedOverState) {
        when (state) {
            DraggedOverState.Left -> {
                isDraggedOverLeft = true
                isDraggedOverRight = false
            }
            DraggedOverState.Right -> {
                isDraggedOverLeft = false
                isDraggedOverRight = true
            }
            DraggedOverState.None -> {
                isDraggedOverLeft = false
                isDraggedOverRight = false
            }
        }
    }

    fun getDraggedOverState(): DraggedOverState {
        when {
            isDraggedOverLeft && !isDraggedOverRight -> return DraggedOverState.Left
            isDraggedOverRight && !isDraggedOverLeft -> return DraggedOverState.Right
            else -> return DraggedOverState.None
        }
    }

    enum class DraggedOverState {
        None,
        Left,
        Right
    }

}
